### ISSUE TITLE
jshint has trouble with embedded php statements

### ISSUE THREAD

**Issue by davellan** [SublimeLinter/SublimeLinter-jshint#85]
For example:

```
    // Quick search enabled search
    <?php if($_POST['Quick']) :?>
    $('#SearchForm').submit();
    <?php endif; ?>
```

This creates 23 errors with no descriptions, a mix of red and yellow highlighting as shown in the image below:
![jshint lint error](https://cloud.githubusercontent.com/assets/12041244/11870675/e00f3f76-a497-11e5-8624-93a3d49aaf66.jpg)

I have these packages installed: docblockr, emmet, sublime linter, gist, autoprefixer, packagecontrol, gotodocumentation,sftp,plaintask.

I have these linters installed: jshint, csslint, php, html-tidy

They all seem to work fine and report errors properly, only the jshint has issues with the embedded php which runs fine on a browser.

**Comment by KasperFranz** [comment]
isn't it the same issue as #35?

**Comment by davellan** [comment]
@KasperFranz - Yes it seems to be. Didn't see it before, so I guess this should be merged with that and closed. That's a shame, as it is annoying for me to see errors reported where there aren't any. :(

**Comment by alexandre1985** [comment]
Me too. @davellan is right. The linter should not throw any error on php

**Comment by origamifreak2** [comment]
I agree, this would help alot.

**Comment by lsl** [comment]
Has anyone done any work on this? Where would be a good place to start if I had some time to spare?

**Comment by braver** [comment]
@lsl you should be able to tweak the selector setting to ignore the php inbetween the javascript. Probably something like `source.js - source.php`. 

- http://www.sublimelinter.com/en/stable/linter_settings.html#selector
- https://www.sublimetext.com/docs/3/scope_naming.html
- http://docs.sublimetext.info/en/latest/extensibility/syntaxdefs.html?highlight=selector#scopes

**Comment by v1xiii** [comment]
Has anyone found an actual solut

> @lsl you should be able to tweak the selector setting to ignore the php inbetween the javascript. Probably something like `source.js - source.php`.
> 
> * http://www.sublimelinter.com/en/stable/linter_settings.html#selector
> * https://www.sublimetext.com/docs/3/scope_naming.html
> * http://docs.sublimetext.info/en/latest/extensibility/syntaxdefs.html?highlight=selector#scopes

I added the following to the sublime linter user settings, no luck. Is that the right place for it?

```
{
    "linters": {
        "jshint":
        {
            "selector": "source.js - source.php"

        },
    }
}
```

**Comment by kaste** [comment]
@v1xiii That looks correct. But this will not make it work bc the javascript your then sending to jshint may be invalid.

Maybe it is better to use eslint with plugins, e.g. https://www.npmjs.com/package/eslint-plugin-php-markup.

Maybe there is a php linter which incorporates js testing as well.
